apps/sync/js/adapters/bookmarks.js

```diff
@@ -305,7 +322,10 @@ DataAdapters.bookmarks = {
         type: payload.type === 'bookmark' ? 'url' : 'others',
         iconable: false,
         icon: '',
-        fxsyncPayload: payload,
+        fxsyncRecord: {
+          last_modified: remoteRecords[i].last_modified,
+          payload: remoteRecords[i].payload
+        },
         fxsyncId: payload.id
       });
     }
```

Comment:
Yes, I didn't refactor the code too much. I think in fact the whole step of constructing the `bookmarks` array, and then looping over it, could probably be skipped?